crates/adapters/src/secret_resolver.rs

```diff
@@ -0,0 +1,387 @@
+use feldera_types::config::ConnectorConfig;
+use feldera_types::secret_ref::{MaybeSecretRef, MaybeSecretRefParseError, SecretRef};
+use serde_yaml::{Mapping, Value};
+use std::fmt::Debug;
+use std::fs;
+use std::path::Path;
+use thiserror::Error as ThisError;
+
+/// Path of the default secrets directory.
+pub const DEFAULT_SECRETS_DIRECTORY_PATH: &str = "/etc/feldera-secrets";
+
+#[derive(Debug, Clone, PartialEq, Eq, ThisError)]
+pub enum SecretRefResolutionError {
+    #[error("{e}")]
+    MaybeSecretRefParseFailed { e: MaybeSecretRefParseError },
+    #[error("secret reference '{secret_ref}' resolution failed: file '{path}' does exist but unable to read it (this might be due to insufficient permissions or content is not UTF-8)")]
+    CannotReadSecretFile { secret_ref: SecretRef, path: String },
```

Comment:
Any reason not to include the underlying error? It would allow the message to be less vague.

Maybe you don't because std::io:Error isn't `Clone`. I've used two workarounds for that in the past: either to include only the `ErrorKind` out of std::io:Error (which is `Clone`) or to convert it to a `String` and include the `String`.

Reply:
I've added the error kind :+1: